src/main/java/pl/edu/pw/mini/ingreedio/api/auth/service/AuthService.java

```diff
@@ -1,98 +1,102 @@
 import pl.edu.pw.mini.ingreedio.api.user.model.User;
-import pl.edu.pw.mini.ingreedio.api.user.repository.UserRepository;
 
 @Service
 @RequiredArgsConstructor
 public class AuthService {
-    private final AuthRepository authRepository;
-    private final UserRepository userRepository;
-
     private final AuthenticationManager authenticationManager;
-    private final JwtService jwtService;
     private final PasswordEncoder passwordEncoder;
+
+    private final AuthInfoRepository authInfoRepository;
+
+    private final JwtService jwtService;
     private final RefreshTokenService refreshTokenService;
     private final JwtClaimsService jwtClaimsService;
     private final RoleService roleService;
 
     @Transactional
-    public User register(RegisterRequestDto request) {
-        User user = User.builder()
-            .displayName(request.displayName())
-            .email(request.email())
-            .build();
-
+    public AuthInfo register(String username, String password, User user)
+        throws UserAlreadyExistsException {
         AuthInfo authInfo = AuthInfo.builder()
-            .username(request.username())
-            .password(passwordEncoder.encode(request.password()))
+            .username(username)
+            .password(passwordEncoder.encode(password))
             .user(user)
             .roles(roleService.getDefaultUserRoles())
             .build();
 
-        userRepository.save(user);
-        authRepository.save(authInfo);
+        try {
+            authInfoRepository.save(authInfo);
+        } catch (DataIntegrityViolationException ex) {
+            throw new UserAlreadyExistsException();
+        }
 
-        return user;
+        return authInfo;
     }
 
     @Transactional
-    public JwtResponseDto refresh(RefreshTokenRequestDto request) {
-        return refreshTokenService.findByToken(request.refreshToken())
-            .map(refreshTokenService::verifyExpirationOfToken)
-            .map(token -> {
-                AuthInfo authInfo = token.getAuthInfo();
-
-                String jwtToken = jwtService.generateToken(jwtClaimsService
-                    .getJwtUserClaimsByAuthInfo(authInfo));
-                RefreshToken refreshToken = refreshTokenService.refreshToken(token);
-                return JwtResponseDto.builder()
-                    .accessToken(jwtToken)
-                    .refreshToken(refreshToken.getToken())
-                    .build();
-            })
-            .orElseThrow(() -> new RuntimeException("Invalid refresh token."));
+    public JwtAuthTokens refresh(RefreshToken token) {
+        AuthInfo authInfo = token.getAuthInfo();
+
+        String jwtToken = jwtService.generateToken(jwtClaimsService
+            .getJwtUserClaimsByAuthInfo(authInfo));
+        RefreshToken refreshToken = refreshTokenService.refreshToken(token);
+        return JwtAuthTokens.builder().accessToken(jwtToken).refreshToken(refreshToken).build();
```

Comment:
This function doesn't check now if the refresh token passed as parameter is valid. Is this intentional?

Reply:
Well it can be improved, you're right. It's an oversight on my part, or more exactly - oversimplification. Instead of checking for token validity during `RefreshTokenService::getToken(String)` it's better to check it in `RefreshTokenService::refresh(RefreshToken)`.

Reply:
Or -- why not both?